### PR TITLE
[Hotfix][Core] Fix 2D discontinuous distance orientation

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
This PR fixes the discontinuous distance orientation in 2D so to make it consistent with the expected 3D behavior, that is to say, the positive distance side is the side the structure normal is pointing to. Note that the modification in the distance algorithm is consistent with the way we use to calculate the normal in 2D simplex geometries (2-node line).

### CODE REVIEW

File: kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp [hunk 1]
```diff
@@ -682,8 +682,8 @@ namespace Kratos
         const Element::GeometryType &rGeometry,
         array_1d<double,3> &rIntObjNormal)
     {
-        rIntObjNormal[0] = rGeometry[0].Y() - rGeometry[1].Y();
```
Comment (loumalouomega) [inline kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp:685]: :)

**Review by loumalouomega — DISMISSED** [review]
2 lines of code, probably 6 h of debugging

**Comment by rubenzorrilla** [comment]
About the FSI tests, I prefer to deactivate as it will be upgraded in #12673.

@Marco1410 @marcnunezc  I required to update the results of the embedded wake tests.

Note: @Marco1410, we need to upgrade the implementation of the `CompressiblePotentialFlow` application tests. Right now it is a bit intricate...

**Comment by marcnunezc** [comment]
> About the FSI tests, I prefer to deactivate as it will be upgraded in #12673.
> 
> @Marco1410 @marcnunezc I required to update the results of the embedded wake tests.
> 
> Note: @Marco1410, we need to upgrade the implementation of the `CompressiblePotentialFlow` application tests. Right now it is a bit intricate...

Hi! If only the orientation is changing, I would expect no change in the test results. I would like to understand what is causing the change in the result. Something is not robust enough (probably the logic on the intersection of geometry and wake). 

I dont want to block your PR, would you agree on bringing back the previous test results and skip the tests or mark them as Expected Failure until I find the issue?

**Comment by rubenzorrilla** [comment]
> > About the FSI tests, I prefer to deactivate as it will be upgraded in #12673.
> > @Marco1410 @marcnunezc I required to update the results of the embedded wake tests.
> > Note: @Marco1410, we need to upgrade the implementation of the `CompressiblePotentialFlow` application tests. Right now it is a bit intricate...
> 
> Hi! If only the orientation is changing, I would expect no change in the test results. I would like to understand what is causing the change in the result. Something is not robust enough (probably the logic on the intersection of geometry and wake).
> 
> I dont want to block your PR, would you agree on bringing back the previous test results and skip the test or mark it as Expected Failure until I find the issue?

Makes perfect sense. I'll have a quick look to it. If I manage to find the issue I'd try to fix it and revert the test results. If not, I'll mark them as expected failure.

**Comment by rubenzorrilla** [comment]
@marcnunezc I think missed one change in the embedded wake definition process (basically to flip it in order to keep the same values). Please confirm that what I done is correct.

**Comment by marcnunezc** [comment]
> @marcnunezc I think missed one change in the embedded wake definition process (basically to flip it in order to keep the same values). Please confirm that what I done is correct.

So are you getting the original test results with this change?

**Comment by marcnunezc** [comment]
> > > @marcnunezc I think missed one change in the embedded wake definition process (basically to flip it in order to keep the same values). Please confirm that what I done is correct.
> > 
> > 
> > So are you getting the original test results with this change?
> 
> Mainly yes. But there are some slight changes, about which I'm not 100% sure if they can be expected or not.
> 
> Edit: I reverted the changes in the json files to ease the tracking of the changes.

I have a better solution to the problem in https://github.com/KratosMultiphysics/Kratos/pull/12690 using the orientation of the wake to decide the sign of the distance values. With this approach, no changes are required in this PR after changing the orientation of the discontinuous process.

**Comment by rubenzorrilla** [comment]
> > > > @marcnunezc I think missed one change in the embedded wake definition process (basically to flip it in order to keep the same values). Please confirm that what I done is correct.
> > > 
> > > 
> > > So are you getting the original test results with this change?
> > 
> > 
> > Mainly yes. But there are some slight changes, about which I'm not 100% sure if they can be expected or not.
> > Edit: I reverted the changes in the json files to ease the tracking of the changes.
> 
> I have an alternative solution to the problem in #12690 using the orientation of the wake to decide the sign of the distance values. With this approach, no changes are required in this PR after changing the orientation of the discontinuous process.

Fine for me. Whatever you consider is surely the best option in this case. :+1: .

**Comment by rubenzorrilla** [comment]
Should be ready now @loumalouomega @marcnunezc .

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**